### PR TITLE
Log 10569 fixes

### PR DESCRIPTION
**Pull request by c-nixon**
This addresses a couple of issues for agents running in high latency, high event volume environments

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -115,7 +115,7 @@ test: test-journald ## Run unit tests
 .PHONY:integration-test
 integration-test: ## Run integration tests using image with additional tools
 	$(eval FEATURES := $(FEATURES) integration_tests)
-	$(DOCKER_JOURNALD_DISPATCH) "--env LOGDNA_INGESTION_KEY=$(LOGDNA_INGESTION_KEY) --env LOGDNA_HOST=$(LOGDNA_HOST) --env RUST_BACKTRACE=full --env RUST_LOG=$(RUST_LOG)" "cargo test $(FEATURES_ARG) --manifest-path bin/Cargo.toml $(TESTS) -- --nocapture --test-threads=$(INTEGRATION_TEST_THREADS)"
+	$(DOCKER_JOURNALD_DISPATCH) "--env LOGDNA_INGESTION_KEY=$(LOGDNA_INGESTION_KEY) --env LOGDNA_HOST=$(LOGDNA_HOST) --env RUST_BACKTRACE=full --env RUST_LOG=$(RUST_LOG)" "cargo test $(FEATURES_ARG) --release --manifest-path bin/Cargo.toml $(TESTS) -- --nocapture --test-threads=$(INTEGRATION_TEST_THREADS)"
```
Comment (jorgebay) [inline Makefile:118]: Is this intended?

File: common/fs/src/tail.rs [hunk 2]
```diff
@@ -244,6 +248,7 @@ impl Tailer {
                 }
 
                 if let Entry::File { data, .. } = entry {
+                    // TODO: check if it's new
```
Comment (jorgebay) [inline common/fs/src/tail.rs:251]: NIT: Leftover?

File: common/http/src/client.rs [hunk 3]
```diff
@@ -140,8 +140,15 @@ impl Client {
     }
 
     fn should_flush(&self) -> bool {
-        self.buffer_bytes >= self.buffer_max_size
-            || self.last_flush.elapsed() > Duration::from_millis(250)
+        if self.buffer_bytes >= self.buffer_max_size {
+            debug!("filled buffer, flushing");
```
Comment (jorgebay) [inline common/http/src/client.rs:144]: Maybe use `trace!()` instead?

File: common/http/src/client.rs [hunk 4]
```diff
@@ -169,12 +176,11 @@ impl Client {
         let buffer = self.buffer.replace(buffer).unwrap();
         let buffer_size = self.buffer_bytes as u64;
         self.buffer_bytes = 0;
-        self.last_flush = Instant::now();
-
         Metrics::http().add_request_size(buffer_size);
         Metrics::http().increment_requests();
         let body = buffer.end().expect("Failed to close ingest buffer");
         self.make_request(IngestBodyBuffer::from_buffer(body)).await;
+        self.last_flush = Instant::now();
```
Comment (jorgebay) [inline common/http/src/client.rs:183]: Nice findings!

**Review by jorgebay — COMMENTED** [review]
Minor comments and questions below.

File: Makefile (hunk 1 above)
Comment (c-nixon) [reply Makefile:118]: Yeh, previously I couldn't cause any issues with the tight_loop test as the integration test binary was running too slow!

**Review by jorgebay — APPROVED** [review]
:shipit:

**Pull request merged**